src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -205,7 +205,28 @@ private BoundExpression ConvertCaseExpression(TypeSymbol switchGoverningType, CS
 
             if (switchGoverningType.IsNullableType() && convertedCaseExpression.Kind == BoundKind.Conversion)
             {
-                constantValueOpt = ((BoundConversion)convertedCaseExpression).Operand.ConstantValue;
+                var operand = ((BoundConversion)convertedCaseExpression).Operand;
+
+                if (operand.ConstantValue.IsIntegral)
+                {
```

Comment:
I dont really understand what you mean by that.To repro this you need a different type as the argument then you have in the case switch. How do you want to do that with enums without casing a CS0266?

This does not compile:
```csharp
// repro.cs
public static class C
{
    public static bool F(long? x)
    {
        switch (x)
        {
            case Foo.X:
                return true;
            default:
                return false;
        }
    }
    public enum Foo : int { X  = 1 }
}
```